content/soda-gettingstarted/ak-sk-using-api.md

```diff
@@ -0,0 +1,154 @@
+weight: 20
+---
+This document describes how to generate Access Key and Secret Key (AK/SK) using APIs. The AK/SK can be used to authenticate with SODA Identity Service for executing any other APIs in SODA, thereafter.
+
+##### AK: access key ID, which is a unique identifier used in conjunction with a secret access key to sign requests cryptographically.
+  
+##### SK: secret access key used in conjunction with an AK to sign requests cryptographically. It identifies a request sender and prevents the request from being modified.
+
+{{% notice info %}}
+**_To use the multicloud service, an Access Key, Secret Key (AK/SK) must be generated before anything else. To do this you can follow the steps below:_**  <br />
+Assuming that the user have access to cURL, the API execution is displayed using cURL commands.
+{{% /notice %}}
+
+
+#### 1. Create the Token for the "admin" User.
+API URI Format : 'http://<HOST:PORT>/identity/v3/auth/tokens'
+
+```bash
+curl --location --request POST 'http://192.168.xx.yy/identity/v3/auth/tokens' \
+--header 'Content-Type: application/json' \
+--data-raw '{
+  "auth": {
+    "identity": {
+      "methods": [
+        "password"
+      ],
+      "password": {
+        "user": {
+          "name": "admin",
+          "domain": {
+            "name": "Default"
+          },
+          "password": "password"
+        }
+      }
+    }
+  }
+}'
+```
+
+Similarly, tokens can be generated for any of the users. To generate token for any specific user, replace the "name": "admin" with the <user-name> of the "user" and the user's corresponding password. 
+
+#### 2. Generate the Scoped Tokens 
+API URI Format : 'http://<HOST:PORT>/identity/v3/auth/tokens'
+
+  ```bash
+  curl --location --request POST 'http://192.168.xx.yy/identity/v3/auth/tokens' \
+  --header 'X-Auth-Token: gAAAAABg1Ybtbnk2NEbyewe_mOILyo5v0PnaTWGoSyMhq75NaWR_L111111111111111111111_I222222222_M3333333333_ICZUHCIDLtU0pIcKGAvN9zJKXaXBQQIf27a3uq2WM8eQroQ-CMc1-riHbCbrs_iFucc2Z4RCGEzxL6YqTvlzBoarjIsx5' \
+  --header 'Content-Type: application/json' \
+  --data-raw '{
+    "auth": {
+      "identity": {
+        "methods": [
+          "token"
+        ],
+        "token": {
+          "id": "gAAAAABg1Ybtbnk2NEbyewe_mOILyo5v0PnaTWGoSyMhq75NaWR_L111111111111111111111_I222222222_M3333333333_ICZUHCIDLtU0pIcKGAvN9zJKXaXBQQIf27a3uq2WM8eQroQ-CMc1-riHbCbrs_iFucc2Z4RCGEzxL6YqTvlzBoarjIsx5"
+        }
+      },
+      "scope": {
+        "project": {
+          "name": "admin",
+          "domain": {
+            "id": "default"
+          }
+        }
+      }
+    }
+  }'
+  ```
+
+Generate the Scoped Tokens for the user. Here, the scoped token for the "admin" user is generated, using the tokens obtained in the Response Header "X-Subject-Token" from Step 1. 
+
+#### 3. Using the token AK/ SK can be generated using the POST API 
+
+  After the Scoped Tokens are generated, ProjecID and UserID can be obtained from them. 
```

Comment:
Scoped token doesn't have UserID, how it would be obtained?

Reply:
There are many ways of generating the scoped tokens for a particular scope of Authorization. 
In the method described here the "Token authentication with scoped authorization" method of token generation is used.
In this "A token object. The token authentication method is used. This method is typically used in combination with a request to change authorization scope." is used.

There are other methods of generating Scoped Authorization tokens, which include a UserId aswell. 
More details can be found here  : https://docs.openstack.org/api-ref/identity/v3/index.html?expanded=create-credential-detail,delete-credential-detail,token-authentication-with-scoped-authorization-detail#credentials

If required this can be discussed further.